tensorflow/python/ops/nn_ops.py

```diff
@@ -1523,7 +1523,7 @@ def _swap_axis(logits, dim_index, last_index):
   # shape.
   output.set_shape(shape)
 
-  return output
+  return tf.identity(output, name=name)
```

Comment:
Instead of adding an identity op, how about plumbing name through _swap_axis() to array_ops.tranpose() above?